src/tesk_core/job.py

```diff
@@ -22,7 +23,11 @@ def run_to_completion(self, poll_interval, check_cancelled, pod_timeout):
         logging.debug("Creating job '{}'...".format(self.name))
         logging.debug(pprint(self.body))
         self.timeout = pod_timeout
-        self.bv1.create_namespaced_job(self.namespace, self.body)
+        try:
+            self.bv1.create_namespaced_job(self.namespace, self.body)
+        except ApiException as ex:
```

Comment:
So can we actually be sure that the ApiException was caused by name non-uniqueness Conflict.
I suppose there is a way to improve this bit by actually checking that you got 409 and doing what you do here, otherwise - if it is a different error, just rethrow it.
Also - is there a way to actually test it?

Reply:
Added a new condition to check the status of the exception and an exception is raised if the status is not 409. Unit test cases are added to the code base.